src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,166 @@
+
+        yieldParams = _yieldParams;
+
+        emit UpdateYieldParams(_yieldParams);
+    }
+
+    /**
+     * @dev Rebalances yield bearing tokens.
+     * @param minRebalanceAmount minimum amount of token to move between underlying balance and yield.
+     * @param maxRebalanceAmount maximum amount of token to move between underlying balance and yield.
+     */
+    function rebalance(uint256 minRebalanceAmount, uint256 maxRebalanceAmount) external {
+        if (maxRebalanceAmount < minRebalanceAmount) {
+            (minRebalanceAmount, maxRebalanceAmount) = (maxRebalanceAmount, minRebalanceAmount);
+        }
+
+        YieldParams storage params = yieldParams;
+        IERC4626 yieldVault = IERC4626(params.yield);
+
+        if (address(yieldVault) == address(0)) {
+            return;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender,
+            "ZkBobCompoundingPool: Rebalance is an operator-called method"
+        );
+
+        uint256 underlyingBalance = IERC20(token).balanceOf(address(this));
+        uint256 buffer = params.buffer;
+
+        uint256 balancesDiff = (underlyingBalance >= buffer) ? underlyingBalance - buffer : buffer - underlyingBalance;
+        balancesDiff = (balancesDiff >= minRebalanceAmount)
+            ? ((balancesDiff <= maxRebalanceAmount) ? balancesDiff : maxRebalanceAmount)
+            : 0;
+        if (balancesDiff <= params.dust) {
+            return;
+        }
+        if (underlyingBalance > buffer) {
+            investedAssetsAmount += balancesDiff;
+            yieldVault.deposit(balancesDiff, address(this));
+            emit Rebalance(0, balancesDiff);
+        } else {
+            investedAssetsAmount -= balancesDiff;
+            yieldVault.withdraw(balancesDiff, address(this), address(this));
+            emit Rebalance(balancesDiff, 0);
+        }
+    }
+
+    /**
+     * @dev Collects accumulated fees and generated yield for the specific collateral.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     * @param minClaimAmount minimum amount of token to claim.
+     * @return Claimed amount.
+     */
+    function claim(uint256 minClaimAmount) external returns (uint256) {
+        YieldParams storage params = yieldParams;
+
+        IERC4626 yieldVault = IERC4626(params.yield);
+
+        if (address(yieldVault) == address(0)) {
+            return 0;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender, "ZkBobCompoundingPool: Claim is an operator-called method"
+        );
+
+        uint256 dust = params.dust;
+        minClaimAmount = minClaimAmount > dust ? minClaimAmount : dust;
+
+        uint256 currentInvestedSharesAmount = yieldVault.balanceOf(address(this));
+
+        uint256 toClaimAmount = yieldVault.convertToAssets(currentInvestedSharesAmount) - investedAssetsAmount;
```

Comment:
My understanding was that we should subtract `dust` here as well, so that it's always kept on the pool balance